src/main/java/Duke.java

```diff
@@ -1,10 +1,41 @@
 public class Duke {
+    private UserInterface ui;
+    private TaskManager tasks;
+    private FileManager fileManager;
+
+    private Duke(String pathname) {
+        this.ui = new UserInterface();
+        ui.printWelcome();
```

Comment:
This is an interesting way of printing the greeting! However, would you consider this as a task Duke's constructor should be doing? I feel that the printing is not essential to the creation of a Duke object. Perhaps consider calling this function after Duke is created?

Reply:
Hi pohlinwei, thank you for taking the time to review my code i really appreciate it. And also sorry for replying so late. I enjoyed reading your feedback a lot. 
I'll take your suggestions into consideration.